crates/ruff_linter/src/rules/flake8_pyi/rules/exit_annotations.rs

```diff
@@ -249,36 +274,172 @@ fn check_positional_args(
     }
 }
 
+/// Determines whether the positional arguments of an `__exit__` or `__aexit__` method
+/// overload are annotated correctly.
+fn check_positional_args_for_overloaded_method(
+    checker: &mut Checker,
+    non_self_positional_args: &[&ParameterWithDefault],
+    kind: FuncKind,
+    parameters_range: TextRange,
+) {
+    fn parameter_annotation_loosely_matches_predicate(
+        parameter: &ParameterWithDefault,
+        predicate: impl FnOnce(&Expr) -> bool,
+        semantic: &SemanticModel,
+    ) -> bool {
+        parameter
+            .parameter
+            .annotation
+            .as_ref()
+            .map_or(true, |annotation| {
+                predicate(annotation) || is_object_or_unused(annotation, semantic)
+            })
+    }
+
+    let semantic = checker.semantic();
+    let Some(Stmt::ClassDef(parent_class_def)) = semantic.current_statement_parent() else {
+        return;
+    };
+
+    // Collect all the overloads for this method into a SmallVec
+    let function_overloads: SmallVec<[&StmtFunctionDef; 2]> = parent_class_def
```

Comment:
It shouldn't be very hot, no. I mainly used a `SmallVec` because it was already being used elsewhere in this file, and it _is_ exceedingly unlikely that anybody _would_ use more than two overloads for `__exit__` or `__aexit__`. Is there a disadvantage to using a SmallVec here?